consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/pki/PkiQbftExtraDataCodecTest.java

```diff
@@ -112,30 +169,70 @@ public void fullyPopulatedDataProducesCorrectlyFormedExtraDataObject() {
-    assertThat(extraData.getCms()).hasValue(cms);
+    assertThat(extraData.getCms()).isEqualTo(Bytes.EMPTY);
+  }
+
+  /*
+   When encoding for blockchain, we ignore commit seals and round number, but we include the CMS
+  */
+  @Test
+  public void encodingForBlockchainShouldIncludeCms() {
+    final Bytes expectedRawDecoding =
+        Bytes.fromHexString(RAW_EXCLUDE_COMMIT_SEALS_AND_ROUND_NUMBER_ENCODED_STRING);
+    final Bytes encoded =
+        bftExtraDataCodec.encodeWithoutCommitSealsAndRoundNumber(getDecodedExtraData(cms));
+
+    assertThat(encoded).isEqualTo(expectedRawDecoding);
+  }
+
+  @Test
+  public void encodingWithoutCommitSealsShouldIncludeCms() {
+    final Bytes expectedRawDecoding = Bytes.fromHexString(RAW_EXCLUDE_COMMIT_SEALS_ENCODED_STRING);
+    final Bytes encoded = bftExtraDataCodec.encodeWithoutCommitSeals(getDecodedExtraData(cms));
+
+    assertThat(encoded).isEqualTo(expectedRawDecoding);
   }
 
   @Test
-  public void encodingExtraDataWithEmptyCmsMatchesKnownRawHexString() {
-    final Bytes expectedRawDecoding = Bytes.fromHexString(EMPTY_CMS_RAW_HEX_ENCODING_STRING);
-    final Bytes encoded = bftExtraDataCodec.encode(getDecodedExtraDataWithEmptyCms());
+  public void encodingWithAllShouldIncludeCms() {
+    final Bytes expectedRawDecoding = Bytes.fromHexString(RAW_ALL_ENCODED_STRING);
+    final Bytes encoded = bftExtraDataCodec.encode(getDecodedExtraData(cms));
 
     assertThat(encoded).isEqualTo(expectedRawDecoding);
   }
 
+  /*
+   When encoding for proposal, we include commit seals and round number, but we ignore the CMS
+  */
   @Test
-  public void encodingMatchesKnownRawHexString() {
-    final Bytes expectedRawDecoding = Bytes.fromHexString(RAW_HEX_ENCODING_STRING);
-    final Bytes encoded = bftExtraDataCodec.encode(getDecodedExtraData(Optional.of(cms)));
+  public void encodingForCreatingCmsProposal() {
+    final Bytes expectedRawDecoding = Bytes.fromHexString(RAW_PROPOSAL_ENCODED_STRING);
+    final Bytes encoded = bftExtraDataCodec.encodeWithoutCms(getDecodedExtraData(cms));
 
     assertThat(encoded).isEqualTo(expectedRawDecoding);
   }
 
-  private static PkiQbftExtraData getDecodedExtraDataWithEmptyCms() {
-    return getDecodedExtraData(Optional.empty());
+  /*
+   When encoding non-pki extra data, we delegate to the regular QBFT encoder
+  */
+  @Test
+  public void encodingQbftExtraData() {
+    final Bytes expectedRawDecoding = Bytes.fromHexString(RAW_QBFT_EXTRA_DATA);
+    final PkiQbftExtraData pkiBftExtraData = getDecodedExtraData(cms);
+    final BftExtraData bftExtraData =
+        new BftExtraData(
+            pkiBftExtraData.getVanityData(),
+            pkiBftExtraData.getSeals(),
+            pkiBftExtraData.getVote(),
+            pkiBftExtraData.getRound(),
+            pkiBftExtraData.getValidators());
+
+    final Bytes encoded = bftExtraDataCodec.encode(bftExtraData);
+
+    assertThat(encoded).isEqualTo(expectedRawDecoding);
   }
 
```

Comment:
Be great to have some tests showing that pki qbft extra data without cms can be handled by the qbft extra data codec to ensure we have binary compatibility.